activerecord/lib/active_record/relation/predicate_builder/association_query_value.rb

```diff
@@ -31,9 +31,8 @@ def primary_key
         end
 
         def convert_to_id(value)
-          case value
-          when Base
-            value._read_attribute(primary_key)
```

Comment:
Is `_read_attribute(:id)` more performant or semantically different from `public_send(:id)`? Is typecasting skipped? Is that important here, since the value will be injected directly back into SQL and likely coerced to the column type?